src/cpp/src/image_generation/flux_fill_pipeline.hpp

```diff
@@ -0,0 +1,267 @@
+        latent = pack_latents(noise, generation_config.num_images_per_prompt, num_channels_latents, height, width);
+
+        return std::make_tuple(latent, proccesed_image, image_latents, noise);
+    }
+
+    std::tuple<ov::Tensor, ov::Tensor> prepare_mask_latents(ov::Tensor mask_image,
+                                                            ov::Tensor processed_image,
+                                                            const ImageGenerationConfig& generation_config,
+                                                            const size_t batch_size_multiplier = 1) override {
+        OPENVINO_ASSERT(m_pipeline_type == PipelineType::INPAINTING, "'prepare_mask_latents' can be called for inpainting pipeline only");
+
+        const size_t vae_scale_factor = m_vae->get_vae_scale_factor();
+        ov::Shape target_shape = processed_image.get_shape();
+
+        // Prepare mask latent variables
+        ov::Tensor mask_condition = m_image_resizer->execute(mask_image, generation_config.height, generation_config.width);
+        std::shared_ptr<IImageProcessor> mask_processor = mask_condition.get_shape()[3] == 1 ? m_mask_processor_gray : m_mask_processor_rgb;
+        mask_condition = mask_processor->execute(mask_condition);
+
+        const ov::Shape processed_image_shape = processed_image.get_shape();
+        const ov::Shape mask_condition_shape = mask_condition.get_shape();
+        float * processed_image_data = processed_image.data<float>();
+        const float * mask_condition_data = mask_condition.data<float>();
+
+        OPENVINO_ASSERT(processed_image_shape[2] == mask_condition_shape[2] &&
+                        processed_image_shape[3] == mask_condition_shape[3],
+                        "Image and mask sizes are incompatible"
+                       );
+        
+        if (processed_image.get_shape()[1] == mask_condition.get_shape()[1])  { // rgb mask
+            for (size_t i = 0; i < processed_image.get_size(); ++i) {
+                processed_image_data[i] *= (1.0f - mask_condition_data[i]);
+                
+            }
+        } else { // gray mask
+            size_t spatial_size = processed_image_shape[2] * processed_image_shape[3];
+            for (size_t c = 0; c < processed_image_shape[1]; ++c) {
+                for (size_t i = 0; i < spatial_size; ++i) {
+                    processed_image_data[c * spatial_size + i] *= (1.0f - mask_condition_data[i]);
+                }
+            }
+        }
+
+        size_t num_channels_latents = m_vae->get_config().latent_channels;
+        size_t height = generation_config.height / vae_scale_factor;
+        size_t width = generation_config.width / vae_scale_factor;
+
+        // Encode the masked image
+        auto encode_start = std::chrono::steady_clock::now();
+        ov::Tensor masked_image_latent = m_vae->encode(processed_image, generation_config.generator);
+        m_perf_metrics.vae_encoder_inference_duration +=
```

Comment:
```suggestion
        m_perf_metrics.vae_encoder_inference_duration =
```
as we don't perform VAE encoding in some other places inside this pipeline